src/microbe_stage/Microbe.Contact.cs

```diff
@@ -997,6 +1020,17 @@ private void CheckEngulfShape()
         */
     }
 
+    /// <summary>
+    ///   Decrease the remaining invulnerability time
+    /// </summary>
+    private void HandleInvulnerabilityDecay(float delta)
+    {
+        if (invulnerabilityDuration > 0)
+        {
+            invulnerabilityDuration -= delta;
```

Comment:
this could allow invulnerabilityDuration to become less than zero, but personally I don't think it's work the extra computation time to check that, since it shouldn't be a problem.